File: JAVA/PROGRAMACION_ANDROID_2/Examen_Tema_V_DMM/app/src/main/java/com/david/examen_tema_v_dmm/MainActivity.java
package com.david.examen_tema_v_dmm;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.ListView;
import android.widget.Toast;

import com.david.examen_tema_v_dmm.services.Buscar;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    // OBJETOS
    ListView lvTareas;
    Buscar buscar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // ENLACE
        lvTareas = (ListView) findViewById(R.id.lvTareas);
        buscar = new Buscar(this, this, lvTareas);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        // VALIDAMOS OPCIONES
        switch(item.getItemId()){
            case R.id.insertar:
                Intent newtarea = new Intent(getApplicationContext(), FieldsActivity.class);
                newtarea.putExtra("opcion", "NUEVA TAREA");
                newtarea.putExtra("op", 1);
                startActivity(newtarea);
                break;
            case R.id.exit:
                finishAffinity();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onResume() {
        super.onResume();
        buscar.buscar();
    }
}

File: KOTLIN/E9-ServiceExample/app/src/main/java/com/example/serviceexample/MainActivity.kt
package com.example.serviceexample

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.core.app.JobIntentService.enqueueWork
import com.example.serviceexample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    val SERVICE_ID = 1001
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    fun buttonClick(view: View) {
        //enqueueWork(this, MyJobIntentService::class.java, SERVICE_ID, intent)

        intent = Intent(this, MyService::class.java)
        startService(intent)
    }

}